app/src/main/java/com/evelynhasama/bucketbuds/InspoFragment.java

```diff
@@ -48,17 +54,134 @@ public InspoFragment() {
-        return inflater.inflate(R.layout.fragment_inspo, container, false);
+        mView = inflater.inflate(R.layout.fragment_inspo, container, false);
+        mRvActivities = mView.findViewById(R.id.rvActivitiesFI);
+        LinearLayout linearLayout = mView.findViewById(R.id.llRandomFI);
+        cgFilter = mView.findViewById(R.id.cgFilters);
+
+        mRandomView = new View(getContext());
+        mRandomView = inflater.inflate(R.layout.item_activity, linearLayout);
+        setRandomView();
+
+        mRvActivities.setLayoutManager(new LinearLayoutManager(getContext()));
+
+        mActivites = new ArrayList<>();
+        mAdapter = new InspoActivitiesAdapter(getContext(), mActivites, getActivity());
+        mRvActivities.setAdapter(mAdapter);
+        // default mi radius is 20
+        radiusFilter = "20";
+        getData();
+
+        cgFilter.setOnCheckedChangeListener(new ChipGroup.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(ChipGroup group, int checkedId) {
+                switch (checkedId) {
+                    case R.id.c10miles:
+                       radiusFilter = "10";
+                       break;
+                    case R.id.c20miles:
+                        radiusFilter = "20";
+                        break;
+                    case R.id.c50miles:
+                        radiusFilter = "50";
+                }
+                getData();
+            }
+        });
+
+        return mView;
+    }
+
+    private void getData() {
+
+        LocationManager locationManager = (LocationManager) getActivity().getSystemService(Context.LOCATION_SERVICE);
+
+        if (locationManager != null) {
+            // check permissions
+            if (ActivityCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+                ActivityCompat.requestPermissions(getActivity(), new String[]{Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION}, REQUEST_PERMISSION_CODE);
+                return;
+            }
+            // getting GPS status
+            if (!locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)){
+                Toast.makeText(getContext(), "Enable GPS", Toast.LENGTH_SHORT).show();
+                return;
+            };
+
+            Location location = locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
+            Double latitude;
+            Double longitude;
+            if (location != null) {
+                latitude = location.getLatitude();
+                longitude = location.getLongitude();
+                mAdapter.clear();
+                IEventAPI[] apis = {SeatGeekHelper.getInstance(), MusementHelper.getInstance(), TicketMasterHelper.getInstance()};
```

Comment:
nit: I would prefer this to be declared outside as a class member that you can override for tests.

Or do we need to call `getInstance` right before trying to `getEvents`?

Reply:
ooh good point. yeah I'll check it out as a class member and see if it works